Client/mods/deathmatch/logic/CClientModelManager.cpp

```diff
@@ -77,14 +70,14 @@ CClientModel* CClientModelManager::FindModelByID(int iModelID)
     return nullptr;
 }
 
-std::vector<CClientModel*> CClientModelManager::GetModelsByType(const eClientModelType type, const unsigned int minModelID)
+std::vector<std::shared_ptr<CClientModel>> CClientModelManager::GetModelsByType(eClientModelType type, const unsigned int minModelID)
 {
-    std::vector<CClientModel*> found;
+    std::vector<std::shared_ptr<CClientModel>> found;
     found.reserve(m_modelCount);
 
     for (int i = minModelID; i < MAX_MODEL_ID; i++)
     {
-        CClientModel* model = m_Models[i];
+        std::shared_ptr<CClientModel> model = m_Models[i];
         if (model && model->GetModelType() == type)
         {
             found.push_back(model);
```

Comment:
I think this loop can be refactored to be a range based for loop.
I didnt do it for whatever reason back then.
If you decide to keep it this way, make sure `std::shared_ptr<CClientModel> model` is a `const std::shared_ptr<CClientModel>&` or even better `const auto& model` here, because theres no need yet to copy the shared ptr (as it's not as cheap as copying a pointer).